packages/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -285,12 +286,25 @@ async function resolveDependenciesOfDependency (
       { defaultTag: ctx.defaultTag, registry: ctx.registries.default }
     )
   )
+
+  // If --fix-lockfile is specifed, only used name and version field in previous lockfile info
+  // Other fields will be regenerated to make sure it is up to date
+  let infoFromLockfile = extendedWantedDep.infoFromLockfile
+  if (ctx.fixLockfile && extendedWantedDep.infoFromLockfile) {
+    infoFromLockfile = {
+      name: extendedWantedDep.infoFromLockfile?.name,
+      version: extendedWantedDep.infoFromLockfile?.version,
+      depPath: extendedWantedDep.infoFromLockfile?.depPath,
+      pkgId: extendedWantedDep.infoFromLockfile?.pkgId,
+    }
+  }
```

Comment:
Maybe instead of ignoring some data from the lockfile here, remove all those fields from the lockfile before passing it to resolveDependencies(). This can be done in supi/install/index.ts and it will allow to reuse `forceFullResolution` as I suggested, so `fixLockfile` will not be needed in resolve-dependencies.